### ISSUE TITLE
vrn.kp.ru (CNAME)

### ISSUE THREAD

**Issue by dimisa-RUAdList** [uBlockOrigin/uAssets#7088]
Firefox 74.0
uBlock Origin 1.25.0
Default + RU AdList

`https://www.vrn.kp.ru/daily/27007.4/4069144/`

The filter RU AdList contains a regular expression that blocks ads on sites that use one-time adservers. As it turned out, in the case of with uBlock Origin for Firefox, sites break down due to CNAME. I am in search of an optimal solution that will prevent the breakdown of sites and will not lead to ad skipping.

**Comment by mapx-** [comment]
does this filter fix the page ?
`@@||kp.cdnvideo.ru^$stylesheet,domain=vrn.kp.ru`

**Comment by dimisa-RUAdList** [comment]
This rule will only solve this specific problem, but what about hundreds of other sites that are affected by regular expression? Perhaps the CNAME lock should not work globally, but as a filter option?

**Comment by mapx-** [comment]
or (at least) `$stylesheet` should be excluded from the CNAME space

**Comment by uBlock-user** [comment]
CNAMEs doesn't have anything to with network requests types like $script or $stylesheet. It pulls the CNAME for the domains if any, from Firefox's cache.

**Comment by mapx-** [comment]
yep, but (I guess) when uBO applies the filters it knows it's about a CNAME domain and could exclude certain types

**Comment by dimisa-RUAdList** [comment]
Right now this problem leads me to have to check all sites from this regular expression: https://hg.adblockplus.org/ruadlist/rev/f362910bc9a0#l1.13 and add exceptions if necessary.

**Comment by gorhill** [comment]
I will look into adding the ability to exclude canonical names from static filtering something like `@@...$cname`
